nixpkgs_review/github.py

```diff
@@ -16,6 +24,11 @@ def get(self, path: str) -> Any:
             req.add_header("Authorization", f"token {self.api_token}")
         return json.loads(urllib.request.urlopen(req).read())
 
+    def pr_comment(self, pr: int, msg: str) -> None:
+        "Post a comment on a PR with nixpkgs-review report"
+        gh_pr = self.nixpkgs.get_pull(pr)
+        gh_pr.create_issue_comment(msg)
```

Comment:
If you just create a simple comment, I would rather prefer to create a HTTP post request using the API directly: https://developer.github.com/v3/issues/comments/

Than we don't need to add external library.

Reply:
One can use https://docs.python.org/3/library/http.client.html to make POST requests to github.

Reply:
I actually spent many hours wrestling with the JSON encoding prior to going down this route.  In particular, the payload for the message requires packing it into a JSON structure like this:
https://github.com/bhipple/dotfiles/blob/master/bin/borg#L20-L25

Something about packing the string `msg` into `{ "data": msg }` made GitHub really unhappy, but they don't give you any error message; just an error code with a msg saying invalid JSON.  Using the library is nice because it handles all the details of that structure for you.

In a follow-up, I'm also planning on adding `nixpkgs-reivew pr 1234 --comment --approve --merge`, which can approve (anyone) and also merge (if token has maintainer permission), just like the script I linked above.  

Since I'm going to hit 3+ endpoints and the JSON encoding is weird (maybe you know how to do it?), and since the library appears to be well-maintained and even actively supports mypy typing everywhere, I think it's reasonable to take a dependency on it rather than re-implementing the requests ourselves with the low-level `urllib`.  What do you think?

Reply:
With the library, the others look like this:

```python
    def pr_comment(self, pr: int, msg: str) -> None:
        "Post a comment on a PR with nixpkgs-review report"
        gh_pr = self.nixpkgs.get_pull(pr)
        gh_pr.create_issue_comment(msg)

    def pr_approve(self, pr: int) -> None:
        gh_pr = self.nixpkgs.get_pull(pr)
        gh_pr.create_review(event="APPROVE")

    def pr_merge(self, pr: int) -> None:
        gh_pr = self.nixpkgs.get_pull(pr)
        gh_pr.merge()
```
Seems nice and simple/clean.  I just have the comment in this PR since I figured it has more straightforward semantics and would be easier to review/merge/integrate first, but if you want I can update the PR to send all 3 for code review at once.

Reply:
Hmm weird, now I'm getting import errors due to the external lib ... can't win :roll_eyes: 

This was working yesterday, but before I dig in too deeply to fix, do you want to use the lib or rewrite in terms of the lower-level urllib?

Reply:
I can have a look later on. Not having a dependency makes future maintenance easier for me.

Reply:
Merging and approving should not be flags of the `pr` command in my opinion but rather a subcommand that can be called from within the `nix-shell`.
Otherwise it would mean you would approve/merge a PR regardless of the review done in nixpkgs-review.

i.e.

```
$ nixpkgs-review pr 84428
....
nix-shell> nixpkgs-review merge
```

or

```
$ nixpkgs-review pr 84428
....
nix-shell> nixpkgs-review approve
```

The current pull request number could be passed over environment variables.